FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -66,30 +75,40 @@ const uploadPostImage = async (serviceCategory, imageUpload) => {
     const downloadURL = await getDownloadURL(imageRef);
     return downloadURL;
   } catch (error) {
-    throw new Error("Error uploading image");
+    throw new Error(`Error uploading image: ${error.message}`);
   }
 };
 
-const createPost = async (formData, imageUpload, userId) => {
+const createPost = async (formData, imageUpload, userId, userData) => {
```

Comment:
I noticed the capitalization for `userId` is inconsistent, so rename `userId` here to `userID`.

```suggestion
const createPost = async (formData, imageUpload, userID, userData) => {
```